ios/brave-ios/Sources/Brave/Frontend/Sync/SyncAddDevice/SyncAddDeviceDetailView.swift

```diff
@@ -0,0 +1,483 @@
+
+  // View is created to create white frame around the QR code
+  // It is done to solve problems scanning QR code with Android devices
+  // Do not change how QR code is presented (size, background and inset)
+  // Or do not try to Brave Logo in the middle
+  private let qrCodeContainerView = UIView().then {
+    $0.backgroundColor = .white
+    $0.layer.cornerRadius = 8
+    $0.layer.cornerCurve = .continuous
+    $0.layer.borderColor = UIColor(braveSystemName: .dividerSubtle).cgColor
+    $0.layer.borderWidth = 1.0
+  }
+
+  private lazy var qrCodeView = SyncAddDeviceQRCodeView(frame: .zero)
+
+  private let codeWordsContainerView = UIView().then {
+    $0.backgroundColor = UIColor(braveSystemName: .containerBackground)
+    $0.layer.cornerRadius = 8
+    $0.layer.cornerCurve = .continuous
+    $0.layer.borderColor = UIColor(braveSystemName: .dividerSubtle).cgColor
+    $0.layer.borderWidth = 1.0
+    $0.isHidden = true
+  }
+
+  private lazy var codewordsView = UILabel().then {
+    $0.font = UIFont.monospacedSystemFont(ofSize: 15.0, weight: UIFont.Weight.regular)
+    $0.lineBreakMode = NSLineBreakMode.byWordWrapping
+    $0.textColor = UIColor(braveSystemName: .textPrimary)
+    $0.textAlignment = .left
+    $0.numberOfLines = 0
+    $0.setContentHuggingPriority(.defaultHigh, for: .vertical)
+  }
+
+  var syncChainCode: String? {
+    return codewordsView.text
+  }
+
+  private let syncAPI: BraveSyncAPI
+
+  required init(syncAPI: BraveSyncAPI) {
+    self.syncAPI = syncAPI
+
+    super.init(frame: .zero)
+
+    axis = .vertical
+    alignment = .center
+
+    createSyncCodeViews(syncAPI: syncAPI)
+
+    doLayout()
+  }
+
+  @available(*, unavailable)
+  required init(coder: NSCoder) {
+    fatalError()
+  }
+
+  private func doLayout() {
+    addArrangedSubview(qrCodeContainerView)
+    addArrangedSubview(codeWordsContainerView)
+
+    qrCodeContainerView.snp.makeConstraints {
+      $0.centerX.equalToSuperview()
+      $0.width.equalToSuperview().multipliedBy(0.75)
+      $0.width.equalTo(qrCodeContainerView.snp.height)
+    }
+
+    codeWordsContainerView.snp.makeConstraints {
+      $0.leading.trailing.equalToSuperview()
+    }
+
+    qrCodeContainerView.addSubview(qrCodeView)
+
+    qrCodeView.snp.makeConstraints {
+      $0.edges.equalToSuperview().inset(12)
+    }
+
+    codeWordsContainerView.addSubview(codewordsView)
+
+    codewordsView.snp.makeConstraints {
+      $0.edges.equalToSuperview().inset(16)
+    }
+  }
+
+  private func createSyncCodeViews(syncAPI: BraveSyncAPI) {
+    qrCodeView = SyncAddDeviceQRCodeView(syncApi: syncAPI)
+
+    codewordsView.do {
+      $0.text = syncAPI.getTimeLimitedWords(fromWords: syncAPI.getSyncCode())
+    }
+  }
+
+  func swapCodeViewType(_ isSyncCodePresented: Bool) {
+    qrCodeContainerView.isHidden = !isSyncCodePresented
+    codeWordsContainerView.isHidden = isSyncCodePresented
+  }
+
+  func refreshSyncCodeViews() {
+    removeArrangedSubview(qrCodeContainerView)
```

Comment:
This doesn't actually remove the views from the stackView. Is it intended or do you mean `qrCodeContainerView.removeFromSuperview`?

Reply:
Yeah this is totally intended, the parent is stackview so this is removing arranged subviews both are containers. And  new code generation will add them again when the timer is reset. Please check the video

https://github.com/brave/brave-core/assets/6643505/8ebceadb-b07d-473a-959a-8296f6e659c2

timer is adjusted for 30 sec for sake showing how it works.